src/jti_background.js

```diff
@@ -61,6 +64,35 @@ function saveTemplates (templateJSON, callback, responseData = null) {
     });
 }
 
+function getDomains (callback) {
+    var domainArray = {};
+    // Get the default domains
+    var domainArray2 = $.map(DefaultDomainList, function (domain, index) {
+        domain.default = true;
+        return domain;
+    });
+    // sort the default domains
+    domainArray2 = utils.sortArrayByProperty(domainArray2, 'name');
+    // Get the custom domains
+    chrome.storage.sync.get(StorageID, function (data) {
+        if (data[StorageID]) {
+            var domains = data[StorageID].options.domains;
+            domainArray = $.map(domains, function (domain, index) {
+                domain.default = false;
+                return domain;
+            });
+            // Sort them
+            domainArray = utils.sortArrayByProperty(domainArray, 'name');
+            // combine so that the default entries are always at top
+            domainArray = domainArray2.concat(domainArray);
+            callback(true, null, domainArray);
+        } else {
```

Comment:
No need for the `else` and you should be able to simplify this to have only one call to the callback.

There are a bunch of variables going on here.
I think you should be able to implement this with only the `DefaultDomainList`, `customDomainList` variables